Tests/Engine/DataFeeds/LiveTradingDataFeedTests.cs

```diff
@@ -823,19 +817,36 @@ public void HandlesCoarseFundamentalData()
                             list = null;
                             yieldedUniverseData = true;
                         }
+                    }
                 }
                 return Enumerable.Empty<BaseData>();
             });
 
+            var dataReachedSelection = false;
+            _algorithm.AddUniverse(new FuncUniverse(
+                new SubscriptionDataConfig(typeof(CoarseFundamental), symbol, Resolution.Daily, TimeZones.NewYork, TimeZones.NewYork, false, false, false),
+                new UniverseSettings(Resolution.Second, 1, true, false, TimeSpan.Zero), SecurityInitializer.Null,
+                coarse =>
+                {
+                    dataReachedSelection = true;
+                    return coarse.Take(10).Select(x => x.Symbol);
+                }
+            ));
 
-            ConsumeBridge(feed, TimeSpan.FromSeconds(5), ts =>
+            ConsumeBridge(feed, TimeSpan.FromSeconds(2), ts =>
             {
                 Assert.IsTrue(_dataManager.DataFeedSubscriptions
                     .Any(x => x.IsUniverseSelectionSubscription), "No universe selection subscriptions found.");
             });
 
             timer.Dispose();
+            for (int i = 0; i < 10; i++)
+            {
+                Assert.IsTrue(_dataManager.DataFeedSubscriptions
+                    .Any(subscription => subscription.Configuration.Symbol.Value == $"{i}"));
```

Comment:
Might be useful to add the description argument to `Assert.IsTrue` to know which iteration fails if it should happen?

Reply:
I'll add a note to the assert 👍